engine/src/main/java/io/zeebe/engine/processing/bpmn/container/ProcessProcessor.java

```diff
@@ -174,8 +169,23 @@ public void onChildTerminated(
 
     if (flowScopeContext.getIntent() == ProcessInstanceIntent.ELEMENT_TERMINATING
         && stateBehavior.canBeTerminated(childContext)) {
+
+      final var parentProcessInstanceKey = flowScopeContext.getParentProcessInstanceKey();
+      BpmnElementContext parentElementInstanceContext = null;
+      if (parentProcessInstanceKey > 0) {
+        // needs to be done before we delete the Process element instance
+        parentElementInstanceContext =
+            stateBehavior.getParentElementInstanceContext(flowScopeContext);
+      }
+
+      // the event appliers will remove the process instance, this is the reason
+      // why we need to extract the parent element instance before the reference is deleted
       stateTransitionBehavior.transitionToTerminated(flowScopeContext);
 
+      if (parentElementInstanceContext != null) {
+        stateTransitionBehavior.onCalledProcessTerminated(
+            flowScopeContext, parentElementInstanceContext);
```

Comment:
Nit: this block is very similar in the onTerminate - maybe it could be dry'd up? Happy to leave that for later.

Reply:
Yeah I thought also long time about that, but it felt not worth to introduce a new method which gets some callbacks etc makes things unnecessary complex I thought